widget/embedded/src/hooks/useSyncStoresWithConfig.ts

```diff
@@ -90,8 +79,19 @@ export function useSyncStoresWithConfig() {
   ]);
 
   useEffect(() => {
-    if (fromToken && fromTokensConfig) {
-      if (!fromTokensConfig.some((token) => tokensAreEqual(token, fromToken))) {
+    if (
+      fromToken &&
+      fromTokensConfig &&
+      fromTokensConfig[fromToken?.blockchain]
+    ) {
+      const hasToken = fromTokensConfig[fromToken?.blockchain].tokens.some(
+        (asset) => tokensAreEqual(asset, fromToken)
+      );
+
+      if (
+        (!isExcluded(fromTokensConfig, fromToken?.blockchain) && !hasToken) ||
+        (isExcluded(fromTokensConfig, fromToken?.blockchain) && hasToken)
+      ) {
         setFromToken({ token: null });
       }
     }
```

Comment:
We could extract some functions to avoid duplicate codes. (I didn't test them, please make sure they are working as expected)


```ts
const handleTokenChange = (token, tokensConfig, setToken, blockchainsConfig) => {
  if (token && tokensConfig && tokensConfig[token?.blockchain]) {
    const hasToken = tokensConfig[token?.blockchain].tokens.some(
      (asset) => tokensAreEqual(asset, token)
    );

    const isExcludedCondition = isExcluded(tokensConfig, token?.blockchain);

    if ((!isExcludedCondition && !hasToken) || (isExcludedCondition && hasToken)) {
      setToken({ token: null });
    }
  }
};

const handleBlockchainChange = (blockchain, blockchainsConfig, setBlockchain) => {
  if (blockchain && blockchainsConfig) {
    if (!blockchainsConfig.includes(blockchain.name)) {
      setBlockchain(null);
    }
  }
};

```

and then use it here:

```ts
useEffect(() => {
  handleTokenChange(fromToken, fromTokensConfig, setFromToken, fromBlockchainsConfig);
  handleBlockchainChange(fromBlockchain, fromBlockchainsConfig, setFromBlockchain);
}, [fromTokensConfig, fromBlockchainsConfig]);

useEffect(() => {
  handleTokenChange(toToken, toTokensConfig, setToToken, toBlockchainsConfig);
  handleBlockchainChange(toBlockchain, toBlockchainsConfig, setToBlockchain);
}, [toTokensConfig, toBlockchainsConfig]);

```

Reply:
fixed